docs/mccleary.md

```diff
@@ -0,0 +1,35 @@
+# nf-core/configs: McCleary Configuration
+
+All nf-core pipelines have been successfully configured for use on the McCleary at Yale University.
+
+To use, run the pipeline with `-profile mccleary`. This will download and launch the [`mccleary.config`](../conf/mccleary.config) which has been pre-configured with a setup suitable for the McCleary cluster. Using this profile, a docker image containing all of the required software will be downloaded, and converted to a Singularity image before execution of the pipeline.
+
+## Below are non-mandatory information e.g. on modules to load etc
+
+Before running the pipeline you will need to load Java 11 using the environment module system on McCleary. You can do this by issuing the commands below:
+
+```bash
+## Load Java 11 environment modules
+module purge
+module load Java/11
+```
+
+## Recommendation on environment variables to set
+
+Although not required, it is recommended to set certain Nextflow specific environment variables to make better use of cluster resources.
+
+```bash
+NXF_OPTS="-Xms500M -Xmx8G"
+NXF_WORK="$PALMER_SCRATCH"
+NXF_SINGULARITY_CACHEDIR="$PALMER_SCRATCH"
+```
```

Comment:
Not sure whether I would recommend users to set up environment variables, sometimes one forgets they are there and it's hard to debug. Especially the 1st one I would not recommend, and I like directly specifying workdirs for each project

Reply:
Perhaps we can change the language around the recommendation? The core reason for recommending `WORK` and `SINGULARITY_CACHEDIR` is to put all the intermediate files (tons of data) into a location that will be automatically cleaned up by YCRC.

The reason for recommending `OPTS` is from https://www.nextflow.io/blog/2021/5_tips_for_hpc_users.html

---

I see no strong reason to keep the `OPTS` recommendation in place as the JVM should use available memory rather than need to be explicitly told how much heap to allocate. The recommendation to use `NXF_WORK=$PALMER_SCRATCH` I think is a strong recommendation if we frame the reason for it rather than just say "we recommend setting these variables" -- what are your thoughts?

Reply:
Sounds good in principle, and it is a preference thing really, I like providing the work directory for each run directly with  the nextflow parameter `-w palmer_scratch/project_x/work` for each project separately, that way I do not have a mix up of work directories and I can go and debug a run if I need to inside the workdir. For the other two I guess the env variable is fine. In any case setting the `-w` parameter overrides the env variable.